compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -3391,8 +3391,19 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     {
```

Comment:
Can't leave an inline comment bc github is dumb, but there's a:

> `// FIXME: DSTs with static alignment should be allowed`

...above for the enum case. 

But we don't allow that for enums, since their tails are required to be sized. Can you tweak that message so it doesn't look like something that someone may be interested in fixing, since it's not really actionable unless we were to tweak enums in general.